packages/sdk/src/types/json.ts

```diff
@@ -161,3 +222,50 @@ export function jsonStringify(
     }
     return JSON.stringify(input, replacerFunction, space);
 }
+
+/**
+ * Stringify, which ensures concordium domain types are unwrapped to their inner type before stringified.
+ * This should be used if you want to manually deserialize the inner property values, as the serialization is irreversible.
+ *
+ * @param value A JavaScript value, usually an object or array, to be converted.
+ * @param bigintFormat Determines how to handle bigints. Can be set to either:
+ * - `'number'`: uses 'json-bigint to safely serialize,
+ * - `'string'`: converts `bigint` to strings
+ * - `undefined`: must be taken care of manually, e.g. in replacer function.
+ * Defaults to 'number'
+ * @param replacer A function that transforms the results.
+ * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
+ */
+export function jsonUnwrapStringify(
+    input: any,
+    bigintFormat: 'number' | 'string' | undefined = 'number',
```

Comment:
Consider making this an enum to make it clearer at the call site what it means.

Reply:
But I'm a little concerned that this parameter is needed in the first place. It feels brittle. Shouldn't it be decided per type how its fields are serialized?

Reply:
Well we need to do something about bigints.. in the Wallet connectors we use Numbers, which is unsafe for very Big integers, so this is a great alternative to that at least.

Besides you can still pass a replacer function to be called for individual types.

I'll add some unit tests to see what happens if you pass a replacer function while specifying a bigint format, to make sure it's consistent.

We can also just use json-bigint in the wallet connectors lib, if you're not a fan of including it as an option here. I just thought it'd be convenient.

Reply:
Changed default to run no default conversion.